builtin/mainmenu/tab_multiplayer.lua

```diff
@@ -234,6 +371,23 @@ local function main_button_handler(tabview, fields, name, tabdata)
 	return false
 end
 
+function check_filters(filters, server)
+	if filters.pvp ~= nil and filters.pvp ~= server.pvp then
+		return false
+	end
+	if filters.creative ~= nil and filters.creative ~= server.creative then
+		return false
+	end
+	if filters.version ~= nil and filters.version ~= server.version then
+		return false
+	end
+	if filters.damage ~= nil and not filters.damage == server.damage then
+		return false
+	end
+	
+	return true
```

Comment:
```lua
return (not filters.pvp or filters.pvp == server.pvp) and
		(not filters.creative or filters.creative == server.creative) and
		(not filters.version  or filters.version  == server.version) and
		(not filters.damage or filters.damage == server.damage)
```

is debatably better

If you don't do the above, then at least remove the `~= nil` and change `not X == Y` to `X ~= Y`